django_date_extensions/fields.py

```diff
@@ -114,12 +139,28 @@ def to_python(self, value):
         if value == 'past':
             return ApproximateDate(past=True)
 
-        if not ansi_date_re.search(value):
-            raise ValidationError('Enter a valid date in YYYY-MM-DD format.')
+        prefix = None
+
+        if isinstance(value, (datetime.date, datetime.datetime)):
+            year, month, day = value.year, value.month, value.day
+        else:
+            prefix_date = prefix_date_re.search(value)
+            prefix_date_reverse = prefix_date_reverse_re.search(value)
```

Comment:
I'm not sure it should parse anything but what it outputs, what is the use for this 'reverse' parsing?